### ISSUE TITLE
Integrating hCpatcha to Gravityforms

### ISSUE THREAD

**Issue by TheoToni** [hCaptcha/hcaptcha-wordpress-plugin#239]
Hello guys,

I downloaded the hcaptcha plugin for Wordpress.
Under Settings -> hcaptcha -> Integrations, I was able to turn on Gravityforms.

Now hcaptcha will be implemented automatically on every Gravity Form I have on my website.

But on some of the forms, I dont want that behaviour. 

Is there a solution for this?

**Comment by e271828-** [comment]
@TheoToni you may be able to use the shortcode instead to do what you want.

**Comment by kagg-design** [comment]
I have added a general functionality allowing you to filter the protection status of the form. V2.8.0 will be published in days.

How to block hCaptcha on a specific form?

The plugin has a filter to block adding and verifying hCaptcha on a specific form. Below is an example of how to block the hCaptcha widget on the Gravity Form with id = 1. The check for the plugin name should be provided against the plugin's slug (directory/main-plugin-file.php).

`
/**
 * Filters the protection status of a form.
 *
 * @param string     $value   The protection status of a form.
 * @param string[]   $plugins Plugin(s) serving the form.
 * @param int|string $form_id Form id.
 *
 * @return bool
 */
function hcap_protect_form_filter( $value, $plugins, $form_id ) {
	if ( ! in_array( 'gravityforms/gravityforms.php', $plugins, true ) ) {
		// The form is not by Gravity Forms plugin.
		return $value;
	}

	if ( 1 !== (int) $form_id ) {
		// The form has id !== 1.
		return $value;
	}

	// Turn off protection for Gravity form with id = 1.
	return false;
}

add_filter( 'hcap_protect_form', 'hcap_protect_form_filter', 10, 3 );
`

The feature is available for Gravity Forms only so far, but will be easily extended to other forms.

**Comment by kagg-design** [comment]
No, we do not plan to add this functionality to settings. You should use the hook.
